src/Microsoft.TestPlatform.Extensions.TrxLogger/Utility/FileHelper.cs

```diff
@@ -289,6 +329,8 @@ private static bool IsReservedFileName(string fileName)
                     return tryMePath;
                 }
 
+                // Sleep a random amount before trying the next iteration
+                Thread.Sleep(rng.Next(1, 150));
```

Comment:
This will actually block the thread for the given time, that seems unnecessary if we are able to reserve the file name.